include/sys/nvpair.h

```diff
@@ -28,7 +28,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/errno.h>
+#include <errno.h>
```

Comment:
The best way to silence these warnings is to add our own `<sys/errno.h>` under `lib/libspl/include/sys/errno.h`.  This path is processed first when searching for headers in the user space build which allows us to avoid these kinds of changes to all the source files.  A simple header which includes <error.h> should do the trick.

```c
#ifndef _LIBSPL_SYS_ERRNO_H
#define _LIBSPL_SYS_ERRNO_H

#include <error.h>

#endif /* _LIBSPL_SYS_ERRNO_H */
```